extensions/amp-bind/amp-bind.md

```diff
@@ -57,52 +57,64 @@ limitations under the License.
 <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
 
 
+- [Introduction](#introduction)
 - [A simple example](#a-simple-example)
 - [How does it work?](#how-does-it-work)
 - [A slightly more complex example](#a-slightly-more-complex-example)
 - [Details](#details)
   - [State](#state)
-      - [AMP.setState()](#ampsetstate)
-  - [Bindings](#bindings)
-    - [Element-specific attributes](#element-specific-attributes)
+    - [Initializing state with `amp-state`](#initializing-state-with-amp-state)
+    - [Updating state with `AMP.setState()`](#updating-state-with-ampsetstate)
   - [Expressions](#expressions)
     - [Differences from JavaScript](#differences-from-javascript)
+    - [Examples](#examples)
     - [Whitelisted functions](#whitelisted-functions)
-    - [BNF-like grammar](#bnf-like-grammar)
+  - [Bindings](#bindings)
+    - [Element-specific attributes](#element-specific-attributes)
 - [Debugging](#debugging)
   - [Warnings](#warnings)
   - [Errors](#errors)
+- [Appendix](#appendix)
+  - [`<amp-state>` specification](#amp-state-specification)
+    - [Attributes](#attributes)
+  - [Custom Built-in Functions](#custom-built-in-functions)
+  - [Deep-merge with `AMP.setState()`](#deep-merge-with-ampsetstate)
+    - [Removing a variable](#removing-a-variable)
+  - [Expression Grammar](#expression-grammar)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-## A simple example
+## Introduction
```

Comment:
100% agree that readers should be able to dive in quickly, however I don't think an uninitiated reader has enough context to read the example straightaway without the introductory sentence.  We could take out the Introduction header and move these sentences above the TOC maybe?